### PR TITLE
Fixed examples/autocorrection.py

### PR DESCRIPTION
**Pull request by 38elements**
Hi thanks for writing python-prompt-toolkit.
This example does not work as intended.

### CODE REVIEW

File: examples/autocorrection.py [hunk 1]
```diff
@@ -29,11 +26,11 @@ def _(event):
...
         b = event.cli.current_buffer
         w = b.document.get_word_before_cursor()
-
-        if w is not None:
-            if w in corrections:
-                b.delete_before_cursor(count=len(w))
-                b.insert_text(corrections[w])
+        if w != '':
+            for word in corrections:
+                if w in word:
```
Comment (jonathanslenders) [inline examples/autocorrection.py]: Hi @38elements,

Thank you for this fix! I appreciate it, but I don't think it's entirely correct. The ``w != ''``-part is okay. But this also replaces substrings of the ``corrections`` dictionary by their value. (Doesn't this replace ``lrd`` by ``world``?)

**Comment by 38elements** [comment]
Thank you for your responses.
I was fixed code.

**Comment by 38elements** [comment]
I am sorry.
I was wrong.
I did not read the following sentence.
`The word "impotr" will be corrected when the user types a space afterwards.`

**Comment by jonathanslenders** [comment]
Don't worry!
(Never hesitate to create an issue when you think something is wrong.)

**Pull request closed without merging**